feat: add CONTRIBUTING guide (#141)
jenkins-infra/docs.jenkins.io: CONTRIBUTING.adoc

If you want to avoid having other people work on the same issue, feel free to add a comment to the issue stating that you're working on it.
If the last comment on the issue is older than a week and there is no PR linked to the issue, you can assume nobody is working on it.
If you need any help, please ask in the xref:contacts[].

[[contacts]]
=== Communication channels

The Jenkins website is largely maintained by the link:https://jenkins.io/sigs/docs/[Jenkins documentation] special interest group.
This group has various communication channels which can be used to discuss contributing to the website:

* link:https://app.gitter.im/#/room/#jenkins/docs:matrix.org[Gitter chat]
* link:https://jenkins.io/sigs/docs/#meetings[Regular meetings]

== Structure of docs.jenkins.io

The `docs.jenkins.io` repository contains two distinct parts:

1. **Antora** (Versioned)
   - link: link:./docs[docs]
   - This part of the site is built using Antora, a static site generator designed for documentation. It is version-controlled, meaning that different versions of Jenkins documentation are maintained.

>    - You can find more information in the link:https://github.com/jenkins-infra/docs.jenkins.io/tree/main/docs#readme[Antora README.md].

Reviewer: ```suggestion
   - You can find more information in the link:./docs/README.adoc[Antora README.adoc].
```

2. **Gatsby** (Non-Versioned)
   - link: link:./site[site]
   - The non-versioned part of the site is built using Gatsby, a React-based framework.

>    - For more details, you can refer to the link:https://github.com/jenkins-infra/docs.jenkins.io/tree/main/site#readme[Gatsby README.md].

Reviewer: ```suggestion
   - For more details, you can refer to the link:./site/README.md[Gatsby README.md].
```

If you found an issue that you would like to work on, you are welcome to go ahead and start contributing.
Please don't ask to be assigned to an issue, we don't assign individuals to issues, just go ahead and start working on it.
If you want to avoid having other people work on the same issue, feel free to add a comment to the issue stating that you're working on it.
If the last comment on the issue is older than a week and there is no PR linked to the issue, you can assume nobody is working on it.
If you need any help, please ask in the xref:contacts[].

[[contacts]]
=== Communication channels

The Jenkins website is largely maintained by the link:https://jenkins.io/sigs/docs/[Jenkins documentation] special interest group.
This group has various communication channels which can be used to discuss contributing to the website:

* link:https://app.gitter.im/#/room/#jenkins/docs:matrix.org[Gitter chat]
* link:https://jenkins.io/sigs/docs/#meetings[Regular meetings]

== Structure of docs.jenkins.io

The `docs.jenkins.io` repository contains two distinct parts:

1. **Antora** (Versioned)

>    - link: link:/docs[/docs]

Reviewer: ```suggestion
   - link: link:./docs/README.adoc[docs]
```

Author: The link to the readme is 2 rows below. This is just the link to the whole section

https://github.com/jenkins-infra/docs.jenkins.io/pull/141/files/9d61d1f3db63475c45ae6ae5f63751f3c392af8f#diff-cf6df8a86bba8fb7539f0419c4cc2c7033ca75beb25a7979548068b5f01d439fR50

Author: let me know if you prefer to link the readme here too

Reviewer: Your original link is broken, and we don't want a slash in the text for the link. I have just edited the suggestion. Please commit the change.

   - This part of the site is built using Antora, a static site generator designed for documentation. It is version-controlled, meaning that different versions of Jenkins documentation are maintained.
   - You can find more information in the link:https://github.com/jenkins-infra/docs.jenkins.io/tree/main/docs#readme[Antora README.md].

2. **Gatsby** (Non-Versioned)

>    - link: link:/site[/site]

Reviewer: ```suggestion
   - link: link:./site/README.md[site]
```

Author: Same here, the link to the readme is 2 rows below

Reviewer: Same here, your link is broken and we don't want a slash in the text. I have just updated the suggestion. Please commit the change.

-----
[source, asciidoc]
----
This is where I would _cite_ some highlighted AsciiDoc code.
----
-----

AsciiDoc has a number of other features that make authoring documentation easier, such as "link:https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#admon-bl[admonition blocks]", which help call out specific sections, including:

[source, asciidoc]
----
NOTE: This is a notice that you should pay attention to!

CAUTION: This is a common mistake!
----

Becomes:

NOTE: This is a notice that you should pay attention to!

CAUTION: This is a common mistake!

There are too many other helpful macros and formatting options to list here, so we recommended that you refer to the link:https://asciidoctor.org/docs/asciidoc-syntax-quick-reference[quick reference] to become more familiar with what is available.

== Adding new content

To add or modify content, follow these guidelines based on the specific section of the documentation.

=== Website structure and ownership

The Jenkins website hosts various content: user and developer documentation, blog, governance materials, pages for special interest groups and sub-projects, etc.
This information is maintained by multiple teams.

> Ownership domains are also defined in the link:./.github/CODEOWNERS[CODEOWNERS] file.

Reviewer: ```suggestion
```

Reviewer: We do not yet have this "./.github/CODEOWNERS" file set up yet.

Some tips for contributors:

* PRs are open to public, and any GitHub user can review changes and provide feedback.
If you are interested to review changes, please just do so (and thanks in advance!). 
No special permissions are needed.
* If you need help with reviews for documentation changes, you can ask in the link:https://app.gitter.im/#/room/#jenkins/docs:matrix.org[Documentation SIG Gitter channel].

[[merging-common]]
=== Merging changes in common areas

Common area process applies when there is no special ownership or process defined.
PRs to common areas can be merged by any _Copy Editor_ once all of the following apply:

* Conversations in the PR are completed OR it is explicit that a reviewer does not block the change (often indicated by line comments attached to an approving PR review, or by using the term "nit", from "nit-picking")
* There are enough approvals
** For trivial changes (typo fixes, minor improvements) - 1 approval from a _Copy Editor_
** For major changes - at least 2 approvals from reviewers.

[[merging-special-areas]]
=== Merging changes in special areas

Special areas are managed by their owners.

> _Copy Editors_ should not merge substantial changes in these areas unless they get explicit sign-off from owners identified in the link:./.github/CODEOWNERS[CODEOWNERS].

Reviewer: ```suggestion
```

Reviewer: We do not have this `./.github/CODEOWNERS` file set up yet.

> :toc:

Reviewer: ```suggestion
```

Reviewer: The first few lines are no longer needed as we are no longer using the Ruby-based framework Awestruct for the build tool in this repo.

> :toc-placement: preamble

Reviewer: ```suggestion
```

> :toclevels: 3

Reviewer: ```suggestion
```

> 

Reviewer: ```suggestion
```

= Contributing to docs.jenkins.io

image:https://badges.gitter.im/jenkinsci/docs.svg[link="https://app.gitter.im/#/room/#jenkins/docs:matrix.org"]

> toc::[]

Reviewer: ```suggestion
```

> 

Reviewer: ```suggestion
```

== Getting Started

To contribute to this repository, follow these steps:

1. link:https://guides.github.com/activities/forking/[Fork this repository] and ensure that you keep only the `main` branch.
2. Create a new branch from your `main` branch and perform your work on this new branch, avoiding direct changes to the `main` branch.
3. Once your work is completed and committed to your branch, push the changes to your forked repository.
4. link:https://help.github.com/articles/creating-a-pull-request-from-a-fork/[Create a pull request (PR)] from your branch to the `main` branch of this repository.

Forking only needs to be done once. After that, you can keep your fork up to date with the upstream repository thanks to *Sync fork*.
You can make further changes to your forked repository using the GitHub website file editor or by working on a local clone, as described below.

> You can also open this project as a https://www.gitpod.io/[Gitpod workspace], which comes pre-configured with all the tools you will need.

Reviewer: ```suggestion
```

Reviewer: We do not have Gitpod configured for this revamped version of the website yet.

> 

Reviewer: ```suggestion
```

* *all* (default target) will run a full build of the site, including `install`, `ui`, and `antora`.
This also downloads and regenerates external resources.
* *clean* will remove all build output and dependencies in preparation for a full rebuild.
* *install* will download external dependencies and resources necessary to build the site.
As an optimization to make iterating on content locally more pleasant, dependencies and resources are not downloaded again unless the `clean` target is called first.
The exception being `all`, which downloads and regenerates external resources (but not download dependencies because they are more bandwidth intensive).
* *ui* will build the site ui.
* *antora* will build the documentation and run a development server on link:http://localhost:5252/[localhost:5252].
* *help* show the help message.

=== Building the Gatsby site

To work on the Gatsby part of the site, navigate to the `site/` directory and follow the instructions in the link:https://github.com/jenkins-infra/docs.jenkins.io/blob/main/site/README.md#installation[Gatsby README.md > Installation].

== Editing content

Most content on this site is written up in the AsciiDoc markup language.

> [#why-asciidoc]

Reviewer: ```suggestion
[[why-asciidoc]]
```
